pages/search.tsx

```diff
@@ -0,0 +1,234 @@
+    }).filter((current: any, index: number, self: any) =>
+        index === self.findIndex((diff: any) => (
+          diff.value === current.value
+        ))
+      )
+
+    return suggestions
+  }
+
+  const updateSearchValue = (inputValue: SearchInputValue) => {
+    setSearchValue(inputValue)
+  }
+
+  const onSubmit = (inputValue: SearchInputValue) => {
+    if (inputValue === '') return
+
+    router.replace(
+      { query: { q: inputValue } },
+      undefined,
+      { shallow: true }
+    );
+  }
+
+  const hasBadge = (type: string) => {
+    if (type === 'event' || type === 'news') {
+      return t(`search.entity_type_${type}`)
+    }
+    return false
+  }
+
+  useEffect(() => {   
+    const validateQuery = () => {
+      if (q === undefined || q === '') {
+        return false
+      }
+      return true
+    }
+    
+    if (!router.isReady || !validateQuery()) return
+
+    updateSearchValue(q)
+
+    getSearch(pageIndex, q, locale).then((data) => {
+      setResults(data.results)
+      setTotal(data.total)
+      setPages(getPages(data.total))
+    })
+
+  }, [router.isReady, q, pageIndex])
+
+  return (
+    <Layout header={nav} footer={footer}>
+      <Head>
+        <title>{t('site_title')}</title>
+      </Head>
+      <article className="search-page">
+        <Container className="container">
+
+          <h1>{t('search.title')}</h1>
+
+          <SearchInput        
+            onSubmit={onSubmit}
+            onChange={updateSearchValue}
+            // getSuggestions={getSuggestions}
+            suggestionLabelField="value"
+            placeholder={t('search.input_placeholder')}
+            label={t('search.input_label')}
+            hideSearchButton
+            value={searchValue}
+          />
+
+          <Button
+            className="search-button"
+            theme="black"
+            onClick={() => onSubmit(searchValue)}
+            style={{marginTop: "20px"}}
+          >
+            {t('search.input_submit')}
+          </Button>
+
+        { results && 
+          <div id="search-results" style={{margin: "40px auto"}}>
+
+            <div className="summary">
+              <h2>{`${t('search.keyword')} "${q}"`}</h2>
+              <p><strong>{total}</strong> {t('search.result_text')}</p>
+            </div>
+
+            {results.map((result: any, key: any) => {
+              return (
+                <ItemBlock 
+                  {...{
+                    result, 
+                    linkText: t('search.item_link_text'),
+                    badgeText: result.type !== undefined ? hasBadge(result.type[0]) : false
+                  }}
+                  key={key} 
+                />
+              )
+            })}
+
+          { pages > 1 && 
+            <Pagination
+              language={locale as any}
+              onChange={(event, index) => {
+                event.preventDefault()
+                setPageIndex(index)
+                setTimeout(() => {
```

Comment:
Is timeout needed here?

Reply:
On some cases scrolling to up when pager changes didn't work. Assuming the list wasn't loaded yet so added little timeout to fix this.